### PR TITLE
More tidying of fast path algorithm

### PR DESCRIPTION
**Pull request by urschrei**
Use built-in cross product instead of "left turn"
Simplify vertex line angle logic
Use accumulated caliper angle as a stopping condition
Use epsilon from num-traits instead of "small number"
Simplify bounds
Add link to original paper

I still don't understand why the magic "100" number works, but at least we're down to a single magic number, and I couldn't trigger a condition that causes it to diverge from the rtree-based result.

If anyone's got an idea for an algorithm that can generate valid polygons, this would be a good candidate for a property-based test:

1. generate two valid polygons 
2. generate their convex hulls 
3. calculate distance using fast path4 
4. compare to rtree-based version

- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- [ ] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

**Comment by rmanoka** [comment]
I tried to understand the `min_convex_poly_dist` fast-path, but couldn't make much progress on it.  Some food for thought:

1. Can we implement the original rotating calipers algo. on a single convex-polygon to iterate over all antipodal vertices without requiring the scalar `T: Float` ?  My gut feeling is yes.
2. If we could do (1), can the above min. distance between two convex poly be done using it as a sub-routine?
3. Would it help to also expose (1) as another algo. since [wiki page](https://en.wikipedia.org/wiki/Rotating_calipers) says it has many applications.

TL;DR:  the tidying up changes look good to me.

**Comment by frewsxcv** [comment]
What's the status of this PR?

**Comment by urschrei** [comment]
Needs a rebase but otherwise ready to go, @rmanoka's suggestions for future improvements notwithstanding.
